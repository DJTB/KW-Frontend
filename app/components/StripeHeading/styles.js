import styled from 'styled-components';
import { transparentize } from 'polished';

import { white, grey } from 'shared/styles/colors';
import { delta } from 'shared/styles/typography';
import { elementGutterHorizontal } from 'shared/styles/layout';

export const Wrapper = styled.div`
  ${elementGutterHorizontal}
`;

export const Heading = styled.h3`
  ${delta}
  margin: 0 0 1.25em;
  color: ${transparentize(0.1, grey)};
  font-weight: 400;
  line-height: 1.15;
  border-bottom: 2px solid ${transparentize(0.75, grey)};
`;

export const Text = styled.span`
  display: inline-block;
  position: relative;
  top: .5em;
  margin-left: .5em;
  padding: 0 .25em;
  background-color: white;
`;

export const Count = styled.strong`
  font-size: 0.9em;
  margin-right: .3em;
  padding: .1em .4em .15em;
  color: ${white};
  background-color: ${transparentize(0.1, grey)};
  border-radius: 2px;
`;
