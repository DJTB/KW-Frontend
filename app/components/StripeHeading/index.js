import React from "react";
import PropTypes from "prop-types";
import { isNumber } from "lodash";

import { white } from "shared/styles/colors";
import { Heading, Text, Count } from "./styles";

StripeHeading.propTypes = {
  text: PropTypes.string.isRequired,
  bgColor: PropTypes.string,
  count: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
};

StripeHeading.defaultProps = {
  count: false,
  bgColor: white, // should match parent/body background color otherwise strike line appears behind text
};

function StripeHeading({ text, count, bgColor }) {
  return (
    <Heading>
      <Text bgColor={bgColor}>
        {isNumber(count) && <Count>{count}</Count>}
        {text}
      </Text>
    </Heading>
  );
}

export default StripeHeading;
