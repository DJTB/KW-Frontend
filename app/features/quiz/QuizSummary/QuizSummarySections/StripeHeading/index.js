import React from 'react';
import PropTypes from 'prop-types';
import { isNumber } from 'lodash';

import { white } from 'common/styles/colors';
import { Heading, Text, Count } from './styles';

StripeHeading.propTypes = {
  text: PropTypes.string.isRequired,
  bgColor: PropTypes.string,
  count: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
};

StripeHeading.defaultProps = {
  count: false,
  // must match parent background color otherwise strike line appears behind text
  bgColor: white[2],
};

function StripeHeading({ text, count, bgColor }) {
  return (
    <Heading>
      <Text bgColor={bgColor}>
        {isNumber(count) && <Count>{count}</Count>}
        {text}
      </Text>
    </Heading>
  );
}

export default StripeHeading;
