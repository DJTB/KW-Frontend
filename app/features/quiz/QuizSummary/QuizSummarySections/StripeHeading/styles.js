import styled from "styled-components";
import { transparentize } from "polished";

import { white, grey } from "common/styles/colors";
import { delta } from "common/styles/typography";

export const Heading = styled.h3`
  ${delta} margin: 0 0 1.25em;
  color: ${transparentize(0.1, grey)};
  font-weight: 400;
  line-height: 1.15;
  border-bottom: 2px solid ${transparentize(0.75, grey)};
`;

export const Text = styled.span`
  display: inline-block;
  position: relative;
  top: 0.5em;
  margin-left: 0.5em;
  padding: 0 0.25em;
  background-color: ${({ bgColor }) => bgColor};
`;

export const Count = styled.strong`
  font-size: 0.9em;
  margin-right: 0.3em;
  padding: 0.1em 0.4em 0.15em;
  color: ${white};
  background-color: ${transparentize(0.1, grey)};
  border-radius: 2px;
`;
