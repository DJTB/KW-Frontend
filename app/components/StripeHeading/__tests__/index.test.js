import "jest-styled-components";
import React from "react";
import { shallow } from "enzyme";
import StripeHeading from "../index";

describe("<StripeHeading />", () => {
  it("should match baseline snapshot", () => {
    expect(shallow(<StripeHeading text="Apprentice" />)).toMatchSnapshot();
  });
  it("should adopt a count prop", () => {
    expect(shallow(<StripeHeading text="Apprentice" count={22} />)).toMatchSnapshot();
  });
  it("should still render a count prop if 0", () => {
    expect(shallow(<StripeHeading text="Apprentice" count={0} />)).toMatchSnapshot();
  });
});
