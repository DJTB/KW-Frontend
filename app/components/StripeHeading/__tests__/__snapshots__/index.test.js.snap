// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StripeHeading /> should adopt a count prop 1`] = `
<styled.h3>
  <styled.span>
    <styled.strong>
      22
    </styled.strong>
    Apprentice
  </styled.span>
</styled.h3>
`;

exports[`<StripeHeading /> should match baseline snapshot 1`] = `
<styled.h3>
  <styled.span>
    Apprentice
  </styled.span>
</styled.h3>
`;

exports[`<StripeHeading /> should still render a count prop if 0 1`] = `
<styled.h3>
  <styled.span>
    <styled.strong>
      0
    </styled.strong>
    Apprentice
  </styled.span>
</styled.h3>
`;
